schema/test/unit/functions/migration_rebuild_project_summary_report_history_test.sql

```diff
@@ -0,0 +1,126 @@
+
+begin;
+select plan(4);
+
+-- Test Setup --
+truncate table cif.form_change,
+  cif.project,
+  cif.project_revision,
+  cif.operator,
+  cif.attachment,
+  cif.project_attachment
+restart identity cascade;
+
+insert into cif.operator(id, legal_name) overriding system value values (1, 'Test Operator');
+
+select cif.create_project(5);
+update cif.form_change
+  set new_form_data=jsonb_build_object(
+      'operatorId', 1,
+      'fundingStreamRfpId', 1,
+      'projectStatusId', 1,
+      'proposalReference', 'TESTREF-' || id,
+      'summary', 'lorem ipsum dolor sit amet adipiscing eli',
+      'projectName', 'Test Project',
+      'totalFundingRequest', 1000000,
+      'sectorName', 'Agriculture',
+      'projectType', 'Carbon Capture',
+      'score', 50
+      )
+  where form_data_table_name='project';
+
+
+-- create has no previous_form_change_id and id=1
+select cif.create_form_change('create', 'project_summary_report', 'cif', 'project_summary_report', '{}', 123, null, 'committed', '[]');
+
+-- all updates have the same previous_form_change_id=1 and form_data_record_id=123 (bug=form_data_record_id clears on commit)
+select cif.create_form_change('update', 'project_summary_report', 'cif', 'project_summary_report', '{"description": "value"}', 123, 1, 'pending', '[]');
+
+select cif.create_form_change('update', 'project_summary_report', 'cif', 'project_summary_report', '{"description": "most recent"}', 123, 1, 'pending', '[]');
+
+select cif.commit_project_revision(1);
```

Comment:
Now that you've resolved the problem with the summary report handler, I'm wondering if we can be certain that the commit handler will clear the form_data_record_id upon commit. Do you think it would be better to simply pass null values for form_data_record_id instead?

Reply:
Great point, thanks for catching this!